Bottom: 7ca14fa448442906267863bedd851e2e645fbc32
Top:    3832ea18ad131f4289866382952c1eab4f46f525
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-13 14:31:34 +0100

gtk/gl: Don't set "magenta" for a NULL color passed to ghid_set_color

The code has a strdup (name), followed by a test for name == NULL. If
name _were_ ever NULL, the strdup would crash, so we can remove this
test.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 8af8a03..71889d8 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -368,14 +368,6 @@ ghid_set_color (hidGC gc, const char *name)
     }
 
   old_name = strdup (name);
-
-  if (name == NULL)
-    {
-      fprintf (stderr, "%s():  name = NULL, setting to magenta\n",
-               __FUNCTION__);
-      name = "magenta";
-    }
-
   gc->colorname = (char *) name;
 
   if (gport->colormap == NULL)
